src/internal/pjsdb/job_iterator_test.go

```diff
@@ -104,15 +103,15 @@ func TestForEachJobTxByFilter(t *testing.T) {
 	t.Run("valid/has_input", func(t *testing.T) {
 		withDependencies(t, func(d dependencies) {
 			expected := make(map[pjsdb.JobID]bool)
-			targetFs := mockFileset(t, d, "/inputs/0.txt", "fake data")
+			targetInput := mockFileset(t, d, "/inputs/0.txt", "fake data")
 			withForEachJob(t, d, expected,
-				pjsdb.IterateJobsRequest{Filter: pjsdb.IterateJobsFilter{HasInput: []byte(fileset.ID(targetFs).HexString())}},
+				pjsdb.IterateJobsRequest{Filter: pjsdb.IterateJobsFilter{HasInput: []byte(fileset.ID(targetInput).HexString())}},
```

Comment:
Only program is hashed that way. input is not needed?

Reply:
the program hash is stored in the database. This filter though is looking for a matching input and we don't currently hash anything else. Thus we have to go off of an id for now. In the future all fileset ids will be a stable hash, so we won't have a problem of many filesets mapping to the same contents.